### PR TITLE
Change/7050 infra validator aware sync

### LINKED ISSUE #7050: Make `InfrastructureValidator` use the synced entities
Recently the changes regarding central syncing were introduced into the code base. This makes validation logic easier as we do not need complicated lookups anymore. A simple `getByUuid` is sufficient.

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #7050

~Review after #7048 is merged and this branch is rebased on development, I had to cherry-pick the commits from the other PR.~ rebased

The existing infra validation is now aware of SORMAS central which means that it relies on the fact, that every infra entity has a globally unique UUID which is synced over all instances. Lookup can therefore be realized through a simple `getReferenceByUuid`. I updated the tests to reflect this.

### CODE REVIEW

**Comment by vidi42** [comment]
Cannot test due to an exception when sending which results in 500 from the receiver side

**Review by vidi42 — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java [hunk 1]
```diff
@@ -189,224 +146,193 @@ private CommunityReferenceDto lookupLocalCommunity(CommunityReferenceDto communi
...
+		InfrastructureBaseFacade<DTO, INDEX_DTO, REF_DTO, CRITERIA> facade,
+		String i18property,
+		String caption,
+		Consumer<REF_DTO> onNoErrors) {
+		if (dto != null) {
+			REF_DTO match = facade.getReferenceByUuid(dto.getUuid());
+			if (match != null) {
+				onNoErrors.accept(match);
+			} else {
+				validationErrors.add(new ValidationErrorGroup(groupNameTag), new ValidationErrorMessage(i18property, caption, dto.getCaption()));
```
[the remark this answers is not in the document]
Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java]: fixed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java [hunk 2]
```diff
@@ -189,224 +146,193 @@ private CommunityReferenceDto lookupLocalCommunity(CommunityReferenceDto communi
...
-		infrastructureData.responsibleCommunity = lookupLocalCommunity(responsibleCommunity);
-		if (responsibleCommunity != null && infrastructureData.responsibleCommunity == null) {
-			unmatchedFields.add(new ValidationErrorMessage(Validations.sormasToSormasResponsibleCommunity, responsibleCommunity.getCaption()));
+	private <DTO extends EntityDto, INDEX_DTO extends Serializable, REF_DTO extends InfrastructureDataReferenceDto, CRITERIA extends BaseCriteria> void validateInfra(
+		InfrastructureDataReferenceDto dto,
+		String groupNameTag,
+		ValidationErrors validationErrors,
+		InfrastructureBaseFacade<DTO, INDEX_DTO, REF_DTO, CRITERIA> facade,
+		String i18property,
+		String caption,
```
Comment (vidi42) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java:159]: The caption argument can be removed

**Review by vidi42 — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java (hunk 2 above)
Comment (JonasCir) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/data/infra/InfrastructureValidator.java]: fixed

**Review by vidi42 — APPROVED** [review]

**Pull request merged**